step/src/main/java/com/google/sps/DatabaseHandler.java

```diff
@@ -31,27 +33,58 @@ private DatabaseHandler() {
   // TODO: call this method when a user first signs up
   // Method for adding a user to the database
   public static void addUserToDatabase(User user) {
-    idMap.put(user.getId(), user);    
+    userMap.put(user.getId(), user);    
   }
 
   // Method for getting a user object using an id
   public static User getUserById(long id) throws NoSuchElementException {
-    if (idMap.containsKey(id)) {
-      return idMap.get(id);    
+    if (userMap.containsKey(id)) {
+      return userMap.get(id);    
     } else {
       throw new NoSuchElementException();  
     }    
   }
 
   // Method for removing a user from our database
   public static void removeUserById(long id) {
-    if (idMap.containsKey(id)) {
-      idMap.remove(id);    
+    if (userMap.containsKey(id)) {
+      userMap.remove(id);    
     }
   }
+ 
+  // Storing a user's notification
+  public static void addNotification(Notification notification) {
+    long userId = notification.getId();
+    
+    // adding the user to the hashmap if they're not already in there
+    if (!notificationMap.containsKey(userId)) {
+      notificationMap.put(userId, new LinkedList<>());      
+    }
+    
+    // retrieving the user's list of notifications, adding the new notification
+    // to it, and adding it to the hashmap
+    LinkedList<Notification> userNotifications = (LinkedList) notificationMap.get(userId);
+    userNotifications.addFirst(notification);
+    notificationMap.put(notification.getId(), userNotifications);
```

Comment:
What's wrong with notificationMap.get(notification,getId()).addFirst(notification);? Or at least the notification.put() seems excessive. Objects are references so I don't think the .put() is necessary.